File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    
   // (1+a)(1+b)(1+c)
    const obj = {}
    let num = 1
    
    clothes.forEach((data) => {
        let [clothes, type] = [...data]
        obj[type] ? obj[type] = obj[type] + 1 : obj[type] =  1
    })
    
    for(let key in obj) {
        num = num * (obj[key] + 1 )   
    }
    
    return num - 1
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dict = {}
    for item, type in clothes:
        if dict.get(type):
            dict[type] = dict[type] + 1
        else:
            dict[type] = 1
            
    for i in dict.keys():
        answer = answer * (dict[i] + 1)
        
        
        
    return answer -1
          
        
        